docs/registry/README.md

```diff
@@ -0,0 +1,193 @@
+# Registry Proposal
+
+## Objective
+
+Design an **initial** version of the Registry that can support discoverability of 
+the different event types that can be consumed from the eventing mesh.
+
+## Requirements
+
+Our design revolves around the following core requirements:
+
+1. We should have a Registry per namespace to enforce isolation.
+2. The Registry should contain the event types that can be consumed from
+the eventing mesh. If an event type is not ready for consumption, we should explicitly indicate so.
+3. The event types stored in the Registry should contain all the required information 
+for a consumer to create a Trigger without resorting to some other OOB mechanism.
+
+## Design Ideas
+
+### EventType CRD
+
+We propose having a namespaced EventType CRD. Here is an example of how a CR would look like:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  name: repopush
+spec:
+  type: repo:push
```

Comment:
An `EventType` may support multiple `types` in this proposal. I think it's a little bit confusing. I spend a few minutes to understand the relationship between `EventType` and `type`. Then I figure out `EventType` is something like a type of event sources, and `type` is something like different kinds of events (e.g. create/update/delete are different events of databases.) that this type of event sources issue. If my understanding is correct, I suggest to change these two terms for easy understanding. My suggestions are: change `EventType` to `EventSourceType` and change type to `eventTypes` or `events` for simplicity.

Reply:
After reading the proposal several times, I think I might misunderstand `EventType`. `EventType` means a specific event in this proposal, is that correct? It's really a confusing word.

Reply:
Thanks for the comment Daisy, and sorry for the confusion.
 
An EventType object is just that, a type of Event that can flow through the system.
Although you will expect that the `name` field of the EventType object to be sufficient to identify it, we need the actual `cloud event type` to perform filtering on Triggers later on (and also the `source`). As that `cloud event type` might not meet k8s naming conventions, we cannot use it in `EventType.name` here. Thus we just have it in `EventType.spec.type`, and we autogenerate `EventType.name` based on it (by stripping invalid chars, e.g.,  `:`).

Does it make more sense?